### PR TITLE
[DEV-1814] change default trading limit

### PR DESCRIPTION
**Pull request by Yannick1712**
#### Pre-Release

- [ ] Check migrations
  - No database related infos (sqldb-xxx)
  - Impact on GS (new/removed columns)
- [ ] Check for linter errors (in PR)
- [ ] Test basic user operations
  - Login/logout
  - Registration
  - Update user data
  - Create/delete payment routes

#### Post-Release

- Test basic user operations
- Monitor application insights log

### CODE REVIEW

**Comment by Yannick1712** [comment]
- Adapt userData volumes
- Adapt UI's

File: src/shared/payment/services/transaction-helper.ts [hunk 1]
```diff
@@ -135,21 +148,31 @@ export class TransactionHelper implements OnModuleInit {
     minFee: number,
     from: Asset | Fiat,
     to: Asset | Fiat,
+    tradingLimit: number | undefined,
   ): Promise<TargetEstimation> {
     const price = await this.priceProviderService.getPrice(from, to);
 
+    const fiatChf = from.name !== 'CHF' ? undefined : await this.fiatService.getFiatByName('CHF');
+    const tradingLimitPrice = from.name !== 'CHF' ? undefined : await this.priceProviderService.getPrice(fiatChf, from);
```
Comment (davidleomay) [inline src/shared/payment/services/transaction-helper.ts:156]: Are you sure this is correct? 😉

File: src/shared/payment/services/transaction-helper.ts [hunk 2]
```diff
@@ -135,21 +148,31 @@ export class TransactionHelper implements OnModuleInit {
...
+      ? currentTradingLimit
+      : tradingLimitPrice.convert(currentTradingLimit * 0.99, -1); // -1% for the conversion
+
     return {
       exchangeRate: this.round(price.price, from instanceof Fiat),
       feeAmount: this.round(feeAmount, from instanceof Fiat),
       estimatedAmount: this.round(targetAmount, to instanceof Fiat),
       sourceAmount: this.round(sourceAmount, from instanceof Fiat),
+      tradingLimit: tradingLimitInSource,
     };
```
Comment (davidleomay) [inline src/shared/payment/services/transaction-helper.ts:176]: It's a bit weird to return the `tradingLimit` in source asset in a method called `getTargetEstimation` 🤔

File: src/shared/payment/services/transaction-helper.ts [hunk 3]
```diff
@@ -160,12 +179,30 @@ export class TransactionHelper implements OnModuleInit {
...
+    tradingLimit?: number | undefined,
+  ): Promise<TxSpec> {
     const price = await this.priceProviderService.getPrice(from, this.eur).then((p) => p.invert());
 
+    const fiatChf =
+      from.name === 'CHF' || tradingLimit == undefined ? undefined : await this.fiatService.getFiatByName('CHF');
+    const tradingLimitPrice =
+      from.name === 'CHF' || tradingLimit == undefined
+        ? undefined
+        : await this.priceProviderService.getPrice(fiatChf, from);
```
Comment (davidleomay) [inline src/shared/payment/services/transaction-helper.ts:194]:
```ts
const tradingLimitPrice = tradingLimit && await this.priceProviderService.getPrice(fiatChf, from);
```

File: src/shared/payment/services/transaction-helper.ts [hunk 4]
```diff
@@ -160,12 +179,30 @@ export class TransactionHelper implements OnModuleInit {
...
+
+    const tradingLimitInSource = !tradingLimit
+      ? undefined
+      : !tradingLimitPrice
+      ? tradingLimit
+      : tradingLimitPrice.convert(tradingLimit * 0.99, -1); // -1% for the conversion
```
Comment (davidleomay) [inline src/shared/payment/services/transaction-helper.ts:200]:
```ts
const tradingLimitPrice = tradingLimit && await this.priceProviderService.getPrice(fiatChf, from);

const tradingLimitInSource = tradingLimit && (from.name === 'CHF'
      ? tradingLimit
      : this.convert(tradingLimit * 0.99, tradingLimitPrice, from instanceof Fiat)); // -1% for the conversion
```

File: src/shared/payment/services/transaction-helper.ts [hunk 5]
```diff
@@ -160,21 +182,37 @@ export class TransactionHelper implements OnModuleInit {
...
     };
   }
 
-  private async convertToTarget(to: Asset | Fiat, { minFee, minVolume }: TxSpec): Promise<TxSpec> {
+  private async convertToTarget(
+    to: Asset | Fiat,
+    { minFee, minVolume, maxVolume }: TxSpecExtended,
+  ): Promise<TxSpecExtended> {
     const price = await this.priceProviderService.getPrice(this.eur, to);
+    const tradingLimitPrice = maxVolume && (await this.priceProviderService.getPrice(this.chf, to));
```
Comment (davidleomay) [inline src/shared/payment/services/transaction-helper.ts:208]: `maxVolumePrice`?

**Review by davidleomay — APPROVED** [review]

**Review by davidleomay — COMMENTED** [review]
Add currency comments to volume fields

File: src/subdomains/core/buy-crypto/routes/buy/dto/buy.dto.ts [hunk 6]
```diff
@@ -18,10 +18,10 @@ export class BuyDto {
   @ApiProperty()
   bankUsage: string;
 
-  @ApiProperty()
+  @ApiProperty({ description: 'volume in chf' })
   volume: number;
 
-  @ApiProperty()
+  @ApiProperty({ description: 'annualVolume in chf' })
   annualVolume: number;
```
Comment (davidleomay) [inline src/subdomains/core/buy-crypto/routes/buy/dto/buy.dto.ts:25]: `Volume in CHF` and `Annual volume in CHF`

File: src/subdomains/core/buy-crypto/routes/crypto-route/dto/crypto-route.dto.ts [hunk 7]
```diff
@@ -17,10 +17,10 @@ export class CryptoRouteDto {
   @ApiProperty({ type: DepositDto })
   deposit: DepositDto;
 
-  @ApiProperty()
+  @ApiProperty({ description: 'volume in chf' })
   volume: number;
 
-  @ApiProperty()
+  @ApiProperty({ description: 'annualVolume in chf' })
   annualVolume: number;
```
Comment (davidleomay) [inline src/subdomains/core/buy-crypto/routes/crypto-route/dto/crypto-route.dto.ts:24]: Same here

File: src/subdomains/core/sell-crypto/route/dto/sell.dto.ts [hunk 8]
```diff
@@ -23,10 +23,10 @@ export class SellDto {
   @ApiProperty({ type: FiatDto })
   currency: FiatDto;
 
-  @ApiProperty()
+  @ApiProperty({ description: 'volume in chf' })
   volume: number;
 
-  @ApiProperty()
+  @ApiProperty({ description: 'annualVolume in chf' })
   annualVolume: number;
```
Comment (davidleomay) [inline src/subdomains/core/sell-crypto/route/dto/sell.dto.ts:30]: Same here

File: src/subdomains/core/statistic/dto/statistic.dto.ts [hunk 9]
```diff
@@ -21,7 +21,7 @@ export class SettingStatus {
 }
 
 export class StatisticDto {
-  @ApiProperty({ type: TotalVolumeStatistic })
+  @ApiProperty({ type: TotalVolumeStatistic, description: 'amount in chf' })
   totalVolume: TotalVolumeStatistic;
```
Comment (davidleomay) [inline src/subdomains/core/statistic/dto/statistic.dto.ts:25]: `Amount in CHF`

File: src/subdomains/generic/user/models/user/dto/user.dto.ts [hunk 10]
```diff
@@ -76,7 +76,7 @@ export class UserDetailDto extends UserDto implements UserDetails {
   @ApiPropertyOptional()
   refFeePercent?: number;
 
-  @ApiPropertyOptional()
+  @ApiPropertyOptional({ description: 'refVolume in eur' })
   refVolume?: number;
```
Comment (davidleomay) [inline src/subdomains/generic/user/models/user/dto/user.dto.ts:80]: Referral volume in EUR

File: src/subdomains/generic/user/models/user/dto/user.dto.ts [hunk 11]
```diff
@@ -91,13 +91,13 @@ export class UserDetailDto extends UserDto implements UserDetails {
...
+  @ApiProperty({ type: VolumeInformation, description: 'buyVolume in chf' })
   buyVolume: VolumeInformation;
 
-  @ApiProperty({ type: VolumeInformation })
+  @ApiProperty({ type: VolumeInformation, description: 'sellVolume in chf' })
   sellVolume: VolumeInformation;
 
-  @ApiProperty({ type: VolumeInformation })
+  @ApiProperty({ type: VolumeInformation, description: 'cryptoVolume in chf' })
   cryptoVolume: VolumeInformation;
```
Comment (davidleomay) [inline src/subdomains/generic/user/models/user/dto/user.dto.ts:101]: `Buy volume in CHF` etc.

**Review by davidleomay — APPROVED** [review]

**Review by davidleomay — APPROVED** [review]

**Pull request closed without merging**